force-app/main/default/classes/DomainBuilder_Test.cls

```diff
@@ -1,201 +1,207 @@
     @IsTest
     private static void noUnneededRecords() {
-
         // Setup & Exercise
         Contact_t con = new Contact_t();            // 1x insert
         Opportunity_t opp = new Opportunity_t();    // 1x insert
 
         new Account_t()                             // 1x insert
-                .add(new Contact_t())               // 1x insert
-                .add(con)                           // 0x insert
-                .add(opp                            // 0x insert
-                     .add(con))                     // 1x insert OpportunityContactRole
-                .persist();                         // 1x setSavepoint
-        
-        // Verify                
-        System.assertEquals(6, Limits.getDmlRows()); 
-        System.assertEquals(1, [SELECT Count() FROM Account]);
-        System.assertEquals(2, [SELECT Count() FROM Contact]);
-        System.assertEquals(1, [SELECT Count() FROM Opportunity]);
-        System.assertEquals(1, [SELECT Count() FROM OpportunityContactRole]);
-    }
+            .add(new Contact_t())                   // 1x insert
+            .add(con)                               // 0x insert
+            .add(opp                                // 0x insert
+                    .add(con))                      // 1x insert OpportunityContactRole
+            .persist();                             // 1x setSavepoint
 
+        // Verify
+        System.assertEquals(6, Limits.getDmlRows());
+        System.assertEquals(1, [SELECT COUNT() FROM Account]);
+        System.assertEquals(2, [SELECT COUNT() FROM Contact]);
+        System.assertEquals(1, [SELECT COUNT() FROM Opportunity]);
+        System.assertEquals(1, [SELECT COUNT() FROM OpportunityContactRole]);
+    }
 
     @IsTest
     private static void happyPath() {
-
+        // Setup & Exercise
         Contact_t ron = new Contact_t().first('Ron').last('Harris');
 
         new Account_t()
-                .name('Acme Corp')
-                .add( new Contact_t() )
-                .add( new Opportunity_t()
-                            .amount(1000)
-                            .closes(2019, 12)
-                            .contact(ron))
-                .persist();
-
-        System.assertEquals(2, [SELECT Count() FROM Account]); // TODO: Should be 1 actually.
-        System.assertEquals(1, [SELECT Count() FROM Opportunity]);
-        System.assertEquals(2, [SELECT Count() FROM Contact]);
-        System.assertEquals(1, [SELECT Count() FROM OpportunityContactRole]);
+            .name('Acme Corp')
+            .add(ron) // .add( new Contact_t() )
+            .add(new Opportunity_t().amount(1000).closes(2019, 12).contact(ron))
+            .persist();
+
+        // Verify
+        // Note: below commented assert is wrong, because when persisting Contact_t ron, the default Contact_t constructor also creates one Account_t, and since we are not adding it as a child of Acme Corp, they default account is created as well
```

Comment:
@rsoesemann kindly review this comments please

Reply:
As I said I don't want to merge that on as those are not crucial needed additions